spec/support/user_helpers.rb

```diff
@@ -135,20 +150,48 @@ def mock_login(admin: false, providers:, provider:)
       user.save
 
       allow_any_instance_of(ApplicationController).to receive(:is_logged_in).and_return(true)
+      # allow_any_instance_of(ApplicationController).to receive(:logged_in?).and_return(true)
       allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)
       allow_any_instance_of(ApplicationController).to receive(:token).and_return(token)
     end
 
-    def visit_with_expiring_token(path)
-      # Mock calls to URS and login Test User
-      token_body = {
-        'access_token'  => 'new_access_token',
-        'token_type'    => 'Bearer',
-        'expires_in'    => 3600,
-        'refresh_token' => 'refresh_token',
-        'endpoint'      => '/api/users/testuser'
+    def mock_valid_acs_responses(admin: false)
+      # We can't actually use a valid SAMLResponse from Launchpad because they are
+      # time sensitive, so we need to mock that it is valid and provides the data we need
+      allow_any_instance_of(OneLogin::RubySaml::Response).to receive(:is_valid?).and_return(true)
+      allow_any_instance_of(SamlController).to receive(:pull_launchpad_cookie).and_return(token_body(admin: admin)['access_token'])
+
+      nams_attributes = {
+        auid: 'test_user',
+        email: 'testuser@example.com'
       }
-      token_response = Cmr::Response.new(Faraday::Response.new(status: 200, body: token_body))
+      allow_any_instance_of(OneLogin::RubySaml::Response).to receive(:attributes).and_return(nams_attributes)
+
+      # UrsClient get_urs_uid_from_nams_auid
+      profile_response = Cmr::Response.new(Faraday::Response.new(status: 200, body: profile_body(admin: admin)))
+      allow_any_instance_of(Cmr::UrsClient).to receive(:get_urs_uid_from_nams_auid).and_return(profile_response)
+    end
+
+    def real_launchpad_login(admin: false, providers: 'MMT_2')
+      mock_valid_acs_responses(admin: admin)
+
+      # Use the provided user or lookup a previously created user by URS UID
+      user = User.find_or_create_by(urs_uid: profile_body(admin: admin)['uid'])
+
+      # This is a setter on the User model, because we're only supplying it
+      # providers it will assign provider_id for us.
+      if Array.wrap(providers).any?
+        user.providers = Array.wrap(providers)
+        user.save
+      end
+
+      # page.driver.browser.post('/saml/acs', SAMLResponse: ENV['launchpad_saml_response'])
+      visit root_path
+      click_on 'Launchpad Test Login'
```

Comment:
a note about what it is that you are clicking on would be nice. 

it was my idea and I was like, "what is he clicking on? just hit the endpoint" lol